server/src/main/java/org/elasticsearch/plugins/BootstrapPlugin.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.plugins;
+
+/**
+ * Bootstrap plugins are not loaded during normal Elasticsearch operation. This
+ * interface exists to make this explicit.
+ */
+public interface BootstrapPlugin {
```

Comment:
I don't understand what advantages a marker interface gives here. A bootstrap plugin's purpose is to add code to the root classloader, so that it can be used by the jdk and other parts of the system that are loaded before plugins are loaded. I don't see a reason why these plugins should have a Plugin class at all?

Reply:
Ah...I think I originally added it, thinking I'd use it to filter out bootstrap plugins, but of course you need to load the plugin to inspect it 🤦. I should have removed it earlier, but it's gone now.